feat(readme): updated readme for nonExclusive flag

Pen7art opened the pull request:
Updated readme with information about the nonExclusive flag

Pen7art commented:
@todbot heres the PR to update the README.

Julusian commented:
I think it would be good to mention in these sections that this property is only respected on macos.

Pen7art commented:
@Julusian I updated the readme to mention macOs. 
I also found a bug in windows where if you set the nonExclusive flag the library used it as a parameter to open the device and as expected the operation failed.
I will try to get more test on windows later meanwhile I added a small fix in this PR.

Julusian commented:
I think that the `hid_darwin_set_open_exclusive` call needs to not be moved to be inside the if statement, it will need its own `#if defined(__APPLE__)` which is fine.
I am worried about the case here of a call `HID.HID(somePath, { nonExclusive: false})` followed by a call `HID.HID(somePath)`. That second call should not be affected by the first, but it will be with `hid_darwin_set_open_exclusive` only being called sometimes.

Pen7art commented:
@Julusian  something like this?
